sfkit/sidecar/server.py

```diff
@@ -0,0 +1,71 @@
+import json
+import os
+import socket
+import subprocess
+
+from sfkit.sidecar.utils import get_sock_path
+from sfkit.utils import constants
+
+
+def handle_client(client: socket.socket):
+    try:
+        while True:
+            data = client.recv(1024)
+            if not data:
+                break
+            try:
+                request = json.loads(data.decode("utf-8"))
+            except json.JSONDecodeError:
+                client.sendall("Invalid JSON format".encode("utf-8"))
+                continue
+
+            study_id = request.get("study_id", "")
+            data_path = os.path.realpath(request.get("data_path", ""))
+
+            if not data_path.startswith(constants.SAFE_DATA_PATH):
+                client.sendall("Invalid data_path".encode("utf-8"))
```

Comment:
```suggestion
                client.sendall(f"data_path should start with {constants.SAFE_DATA_PATH}".encode("utf-8"))
```
Might be good to give them feedback on what to expect